### PR TITLE
Salma wants to merge the working project files to the master

### PR DESCRIPTION
**Pull request by salmafg**
#77

@Fattouh92 
@nohahesham 
@mostafahassaan

### CODE REVIEW

File: arability/app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  def index
+    @projects = Project.all
+  end
+
+# author:
+#      Salma Farag
+# description:
+#     A method that calls method createproject that creates the project and redirects to the
+#project page and prints an error if the data entered is invalid
+# params:
```
Comment (mostafahassaan) [inline arability/app/controllers/projects_controller.rb:11]: add ":projects" to params

File: arability/app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      if @project.save
+        format.html { redirect_to "/projects", notice: 'Project was successfully created.' }
+        format.json { render json: @project, status: :created, location: @project }
+      else
+        format.html { render action: "new" }
+        format.json { render json: @project.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+  def show
```
Comment (mostafahassaan) [inline arability/app/controllers/projects_controller.rb:30]: No documentation for this method "def show"

File: arability/app/models/project.rb [hunk 3]
```diff
@@ -1,4 +1,34 @@
...
+  validates :minAge, :presence => true, :inclusion => { :in => 9..99, :message => "is not in range" }
+  validates :maxAge, :presence => true, :inclusion => { :in => 10..100, :message => "is not in range" }, :numericality => { :only_integer => true,:greater_than_or_equal_to => :minAge}
+
+# author:
+#      Salma Farag
+# description:
+#      Takes the params of the project entred by the developer and creates a project compares
+#it to the already existing categories and returns the project
+# params:
+#     parameters of a project
```
Comment (mostafahassaan) [inline arability/app/models/project.rb:16]: change "params" to something more descriptive like "projects" . and then do it like "projects: exemplification..."

File: arability/app/models/project.rb [hunk 4]
```diff
@@ -1,4 +1,34 @@
...
+# success:
+#     Creates and returns a project after splitting the csv categories string and creating
+#new categories and inserting them into the project categories array
+# failure:
```
Comment (mostafahassaan) [inline arability/app/models/project.rb:20]: what does it return if it fails?

File: arability/app/controllers/projects_controller.rb (hunk 1 above)
Comment (salmafg) [inline arability/app/controllers/projects_controller.rb:11]: Hwa msh bya5od parameters asln

File: arability/app/models/project.rb (hunk 3 above)
Comment (salmafg) [inline arability/app/models/project.rb:16]: Msh fahma??

File: arability/app/models/project.rb (hunk 4 above)
Comment (salmafg) [inline arability/app/models/project.rb:20]: There should not be a failure here, sorry. Ha5alleeha none.

**Pull request merged**